### DISCUSSION TITLE
How can I transfer python code to DJL?

### DISCUSSION THREAD

**Discussion by Tomy20Wang** [deepjavalibrary/djl#2482] [Q&A]
Hi! I just started learning neural networks. I am trying to build a network using DJL, but I do not know how to start it, and I have read some DJL demos, but my problem is it is hard to do this for just knowing the python code. And I do not know the usage of Translator or Criteria. Can you give me some advice? Thank you!

**Comment by frankfliu** [comment]
I suggest you start with pre-trained model and run inference in DJL. You can train your model in PyTorch and run inference in DJL.

DJL has many built-in Translators for a known tasks (e.g image classification, text classification, question answering ...), try to use built-in Translator and you can use built-in Translator as base to build your own translators.

You can read this book if you want to learn basics of deeplearning in java: https://d2l.djl.ai/
